openHarmony/classes/oNode.js

```diff
@@ -108,9 +108,100 @@ function oNode ( path, oSceneObject ){
 
   this._type = 'node';
 
-  this.refreshAttributes();
+  // Lazy loading: attributes will be loaded on first access
+  // This dramatically improves performance when creating many node wrappers
+  this._attributes_cached = null;
+  this._attributeGettersCreated = false;
 }
 
+
+// Flag to track if attribute placeholders have been initialized
+oNode._placeholdersInitialized = false;
+
+/**
+ * Initialize placeholder getters for all attribute names across all node types.
+ * This runs ONCE at first node creation, scanning all node types to discover
+ * all possible attribute names, then defines placeholder getters on the prototype.
+ * This allows shorthand access (node.position.x) to work immediately while
+ * still deferring the expensive attribute loading until first use.
+ * @private
+ */
+oNode._initAttributePlaceholders = function() {
+  if (oNode._placeholdersInitialized) return;
+  oNode._placeholdersInitialized = true;
+
+  var allKeywords = {};
+
+  // Guard the scan so temp nodes are always cleaned up
+  var tempGroupPath = null;
+  try {
+    // Get all node types dynamically from Harmony
+    var nodeTypeList = node.getNodeTypeList();
```

Comment:
is this a recent addition? I don't have this function available in harmony 24

Reply:
Oh, this is actually very strange, I can see this function in harmony local documentation (through `Help` menu), while it's not in the online doc... And it works in the API. 😱